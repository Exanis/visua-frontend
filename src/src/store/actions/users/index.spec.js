import * as Actions from './index';

describe('Users Actions', () => {
    it('getUsersList', () => {
        expect(Actions.getUsersList(1, 'test')).toMatchSnapshot();
    });
    it('onUsersList', () => {
        expect(Actions.onUsersList(['first', 'second'])).toMatchSnapshot();
    });
    it('createUser', () => {
        expect(Actions.createUser('test-username', 'test-firstname', 'test-lastname', 'test-email', 'test-password', false, 1, 'test')).toMatchSnapshot();
    });
    it('onCreateUserSuccess', () => {
        expect(Actions.onCreateUserSuccess()).toMatchSnapshot();
    });
    it('onCreateUserError', () => {
        expect(Actions.onCreateUserError(['error'])).toMatchSnapshot();
    });
    it('resetCreateUserState', () => {
        expect(Actions.resetCreateUserState()).toMatchSnapshot();
    });
    it('editUser', () => {
        expect(Actions.editUser('test-uuid', 'test-username', 'test-firstname', 'test-lastname', 'test-email', 'test-password', false, 1, 'test')).toMatchSnapshot();
    });
    it('onEditUserSuccess', () => {
        expect(Actions.onEditUserSuccess()).toMatchSnapshot();
    });
    it('onEditUserError', () => {
        expect(Actions.onEditUserError(['error'])).toMatchSnapshot();
    });
    it('resetEditUserState', () => {
        expect(Actions.resetEditUserState()).toMatchSnapshot();
    });
    it('deleteUser', () => {
        expect(Actions.deleteUser('test-uuid', 1, 'test')).toMatchSnapshot();
    });
    it('onDeleteUserSuccess', () => {
        expect(Actions.onDeleteUserSuccess()).toMatchSnapshot();
    });
    it('resetDeleteUserState', () => {
        expect(Actions.resetDeleteUserState()).toMatchSnapshot();
    });
});