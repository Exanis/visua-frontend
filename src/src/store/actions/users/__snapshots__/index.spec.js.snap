// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Actions createUser 1`] = `
Object {
  "email": "test-email",
  "firstName": "test-firstname",
  "isStaff": false,
  "lastName": "test-lastname",
  "page": 1,
  "password": "test-password",
  "search": "test",
  "type": "users.create",
  "username": "test-username",
}
`;

exports[`Users Actions deleteUser 1`] = `
Object {
  "page": 1,
  "search": "test",
  "type": "users.delete",
  "uuid": "test-uuid",
}
`;

exports[`Users Actions editUser 1`] = `
Object {
  "email": "test-email",
  "firstName": "test-firstname",
  "isStaff": false,
  "lastName": "test-lastname",
  "page": 1,
  "password": "test-password",
  "search": "test",
  "type": "users.edit",
  "username": "test-username",
  "uuid": "test-uuid",
}
`;

exports[`Users Actions getUsersList 1`] = `
Object {
  "page": 1,
  "search": "test",
  "type": "users.list",
}
`;

exports[`Users Actions onCreateUserError 1`] = `
Object {
  "errors": Array [
    "error",
  ],
  "type": "users.create.error",
}
`;

exports[`Users Actions onCreateUserSuccess 1`] = `
Object {
  "type": "users.create.ok",
}
`;

exports[`Users Actions onDeleteUserSuccess 1`] = `
Object {
  "type": "users.delete.ok",
}
`;

exports[`Users Actions onEditUserError 1`] = `
Object {
  "errors": Array [
    "error",
  ],
  "type": "users.edit.error",
}
`;

exports[`Users Actions onEditUserSuccess 1`] = `
Object {
  "type": "users.edit.ok",
}
`;

exports[`Users Actions onUsersList 1`] = `
Object {
  "type": "users.onList",
  "users": Array [
    "first",
    "second",
  ],
}
`;

exports[`Users Actions resetCreateUserState 1`] = `
Object {
  "type": "users.create.reset",
}
`;

exports[`Users Actions resetDeleteUserState 1`] = `
Object {
  "type": "users.delete.reset",
}
`;

exports[`Users Actions resetEditUserState 1`] = `
Object {
  "type": "users.edit.reset",
}
`;
